clientconn.go

```diff
@@ -129,6 +132,13 @@ func WithBalancer(b Balancer) DialOption {
 	}
 }
 
+// WithServiceConfig returns a DialOption which has a channel to read the service configuration.
+func WithServiceConfig(c <-chan ServiceConfig) DialOption {
```

Comment:
Apparently using custom struct is a better way to deliver the signal to our users that grpc does not put any restrictions on the format you use to transmit service config. Plus, if you choose custom struct, you gain the benefit avoiding unnecessary conversions (yes, it is not approach killer but it is a benefit.). 

I do not understand why any benefit you mentioned in the 2nd paragrahp cannot be achieved with my approach. Again, the approach here does not exclude JSON impl and is a superset -- it just does not fuse the API with JSON.

Not available today. will schedule some time to talk to you next Monday. Thanks.